test/Unit/SqlTriggerBindingTests.cs

```diff
@@ -0,0 +1,94 @@
+            var configProvider = new SqlTriggerAttributeBindingProvider(config.Object, hostIdProvider.Object, loggerFactory.Object);
+            await Assert.ThrowsAsync<ArgumentNullException>(() => configProvider.TryCreateAsync(null));
+        }
+
+        [Fact]
+        public void TestTriggerListenerNullConfig()
+        {
+            string connectionString = "testConnectionString";
+            string tableName = "testTableName";
+            string userFunctionId = "testUserFunctionId";
+
+            Assert.Throws<ArgumentNullException>(() => new SqlTriggerListener<TestData>(null, tableName, userFunctionId, mockExecutor.Object, logger.Object));
+            Assert.Throws<ArgumentNullException>(() => new SqlTriggerListener<TestData>(connectionString, null, userFunctionId, mockExecutor.Object, logger.Object));
+            Assert.Throws<ArgumentNullException>(() => new SqlTriggerListener<TestData>(connectionString, tableName, null, mockExecutor.Object, logger.Object));
+            Assert.Throws<ArgumentNullException>(() => new SqlTriggerListener<TestData>(connectionString, tableName, userFunctionId, null, logger.Object));
+            Assert.Throws<ArgumentNullException>(() => new SqlTriggerListener<TestData>(connectionString, tableName, userFunctionId, mockExecutor.Object, null));
+        }
+
+        [Fact]
+        public void TestTriggerBindingNullConfig()
+        {
+            string connectionString = "testConnectionString";
+            string tableName = "testTableName";
+
+            Assert.Throws<ArgumentNullException>(() => new SqlTriggerBinding<TestData>(null, connectionString, TriggerBindingFunctionTest.GetParam(), hostIdProvider.Object, logger.Object));
+            Assert.Throws<ArgumentNullException>(() => new SqlTriggerBinding<TestData>(tableName, null, TriggerBindingFunctionTest.GetParam(), hostIdProvider.Object, logger.Object));
+            Assert.Throws<ArgumentNullException>(() => new SqlTriggerBinding<TestData>(tableName, connectionString, null, hostIdProvider.Object, logger.Object));
+            Assert.Throws<ArgumentNullException>(() => new SqlTriggerBinding<TestData>(tableName, connectionString, TriggerBindingFunctionTest.GetParam(), null, logger.Object));
+            Assert.Throws<ArgumentNullException>(() => new SqlTriggerBinding<TestData>(tableName, connectionString, TriggerBindingFunctionTest.GetParam(), hostIdProvider.Object, null));
+        }
+
+        [Fact]
+        public async void TestTriggerBindingProviderTryCreateAsync()
+        {
+            var triggerBindingProviderContext = new TriggerBindingProviderContext(TriggerBindingFunctionTest.GetParam(), cancellationTokenSource.Token);
+            var triggerAttributeBindingProvider = new SqlTriggerAttributeBindingProvider(config.Object, hostIdProvider.Object, loggerFactory.Object);
+
+            await Assert.ThrowsAsync<InvalidOperationException>(() => triggerAttributeBindingProvider.TryCreateAsync(triggerBindingProviderContext));
```

Comment:
Make sure that the test name indicates what is being tested - and if needed more details about what to expect as comments in the test code itself. For example for this I don't know what it means to test "TryCreateAsync" and why it's expected to throw here.